src/main/java/se/kth/meta/db/Dbao.java

```diff
@@ -427,6 +423,44 @@ public int getLastInsertedTupleId() throws DatabaseException {
     return query.getResultList();
   }
 
+  /**
+   * Find the Template that has <i>templateid</i> as id.
+   * <p>
+   * @param templateid
+   * @return
+   */
+  public Templates findTemplateById(int templateid) {
+    TypedQuery<Templates> query = em.createNamedQuery(
+            "Templates.findByTemplateid",
+            Templates.class);
+
+    query.setParameter("templateid", templateid);
+    return query.getSingleResult();
+  }
+
+  /**
+   * Update the relationship table <i>meta_template_to_inode</i>
+   * <p>
+   * @param template
+   * @throws se.kth.meta.exception.DatabaseException
+   */
+  public void updateTemplatesInodesMxN(Templates template) throws
+          DatabaseException {
+    try {
+      this.utx.begin();
```

Comment:
We tend to use JTA; since you seem to have method-wide transactions, I don't see a reason for specifying the beginning and end of a transaction yourself. (Calling em.flush() should get you the ID to return.)